app/core/store/modules/classrooms.js

```diff
@@ -0,0 +1,69 @@
+import classroomsApi from 'core/api/classrooms'
+
+export default {
+  namespaced: true,
+
+  state: {
+    loading: {
+      byTeacher: {}
+    },
+
+    classrooms: {
+      // Classrooms by teacher ID
+      //  {
+      //     active: [],
+      //     archived: []
+      //  }
+      byTeacher: {}
+    }
+  },
+
+  mutations: {
+    toggleLoadingForTeacher: (state, teacherId) => {
+      let loading = true
+      if (state.loading.byTeacher[teacherId]) {
+        loading = false
+      }
+
+      Vue.set(state.loading.byTeacher, teacherId, loading)
+    },
+
+    addClassroomsForTeacher: (state, { teacherId, classrooms }) => {
+      const teacherClassroomsState = {
+        active: [],
+        archived: []
+      }
+
+      classrooms.forEach((classroom) => {
+        if (classroom.archived) {
+          teacherClassroomsState.archived.push(classroom)
+        } else {
+          teacherClassroomsState.active.push(classroom)
+        }
+      })
+
+      Vue.set(state.classrooms.byTeacher, teacherId, teacherClassroomsState)
+    }
+  },
+
+  actions: {
+    fetchClassroomsForTeacher: ({ commit }, teacherId) => {
```

Comment:
Could this action be simplified by making it async await syntax?

Reply:
Almost certainly - added a follow up task to track this https://app.asana.com/0/1115428071773315/1118364384628020/f